### PR TITLE
introduce certopts

### PR DESCRIPTION
**Pull request by i0rek**
I liked `CAOpts` from https://github.com/hashicorp/consul/pull/9585 so much that I added `CertOpts` as well.

### CODE REVIEW

File: agent/routine-leak-checker/leak_test.go [hunk 1]
```diff
@@ -15,31 +15,20 @@ import (
...
 
 func testTLSCertificates(serverName string) (cert string, key string, cacert string, err error) {
-	ca, _, err := tlsutil.GenerateCA(tlsutil.CAOpts{})
-	if err != nil {
-		return "", "", "", err
-	}
-
-	// generate leaf
-	serial, err := tlsutil.GenerateSerialNumber()
+	signer, _, err := tlsutil.GeneratePrivateKey()
```
Comment (i0rek) [inline agent/routine-leak-checker/leak_test.go:18]: I think I forgot to fix this one in `CAOpts`: the same signer needs to be used for CA and certs.

**Review by dnephin — APPROVED** [review]
LGTM!

**Comment by freddygv** [comment]
@i0rek is there anything else you would like to add here?

**Comment by i0rek** [comment]
@freddygv I revisited the PR and did it all over again. Should be good now.

**Pull request merged**